libpromises/generic_agent.c

```diff
@@ -518,19 +518,36 @@ static bool LoadAugmentsData(EvalContext *ctx, const char *filename, const JsonE
                     ref->scope = xstrdup("def");
                 }
 
-                const JsonElement *var_info = JsonObjectGet(variables, vkey);
-                const JsonElement *data = JsonObjectGet(var_info, AUGMENTS_VARIABLES_DATA);
+                const JsonElement *const var_info = JsonObjectGet(variables, vkey);
 
-                if (data == NULL)
+                const JsonElement *data;
+                StringSet *tags;
+                const char *comment = NULL;
+
+                if (JsonGetType(var_info) == JSON_TYPE_OBJECT)
+                {
+                    data = JsonObjectGet(var_info, AUGMENTS_VARIABLES_DATA);
+
+                    if (data == NULL)
+                    {
+                        Log(LOG_LEVEL_ERR, "Missing value for the augments variable '%s' in '%s' (value field is required)",
+                            vkey, filename);
+                        continue;
+                    }
+
+                    const JsonElement *json_tags = JsonObjectGet(var_info, AUGMENTS_VARIABLES_TAGS);
+                    tags = GetTagsFromAugmentsTags("variable", vkey, json_tags, "source=augments_file", filename);
+                    comment = GetAugmentsComment("variable", vkey, filename, var_info);
+                }
+                else
                 {
-                    Log(LOG_LEVEL_ERR, "Missing value for the augments variable '%s' in '%s'",
-                        vkey, filename);
-                    continue;
+                    // Just a bare value, like in "vars", no metadata
+                    data = var_info;
+                    tags = GetTagsFromAugmentsTags("variable", vkey, NULL, "source=augments_file", filename);
                 }
 
-                const JsonElement *json_tags = JsonObjectGet(var_info, AUGMENTS_VARIABLES_TAGS);
-                StringSet *tags = GetTagsFromAugmentsTags("variable", vkey, json_tags, "source=augments_file", filename);
-                const char *comment = GetAugmentsComment("variable", vkey, filename, var_info);
+                assert(tags != NULL);
+                assert(data != NULL);
```

Comment:
So leaking tags here as well and it is interesting that the code is duplicated between the two. Maybe a refactor TODO another time? `// TODO` ?

Reply:
We already have a ticket for that.

Reply:
No leak here, since there is no continue here, like in the other file. Also there is already a ticket to refactor this.

Reply:
But I don't think there's a leak possible here.